xtask/src/commands/release/post_release.rs

```diff
@@ -69,6 +70,130 @@ pub fn post_release(workspace: &std::path::Path) -> Result<()> {
                 migration_file_path.display()
             );
         }
+
+        // Backport branch management: only for minor/major bumps on stable releases (**major** >= 1),
+        // we do not have and do not maintain backport branches for `0.{minor}.x` versions.
+        match package_plan.bump {
+            VersionBump::Minor | VersionBump::Major => {
+                if version.major >= 1 {
+                    let branch_name = format!("{}-{}.{}.x", package, version.major, version.minor);
+
+                    // Check if branch already exists on origin
+                    let branch_exists = Command::new("git")
+                        .args(["ls-remote", "--exit-code", "--heads", "origin", &branch_name])
+                        .current_dir(workspace)
+                        .status()
+                        .map(|status| status.success())
+                        .unwrap_or(false);
+
+                    if branch_exists {
+                        log::info!(
+                            "Backport branch already exists on origin: {branch_name}, skipping creation"
+                        );
+                    } else {
+                        let create_status = Command::new("git")
+                            .arg("branch")
+                            .arg(&branch_name)
+                            .current_dir(workspace)
+                            .status()
+                            .with_context(|| {
+                                format!("Failed to create backport branch {branch_name}")
+                            })?;
+
+                        if !create_status.success() {
+                            return Err(anyhow::anyhow!(
+                                "`git branch {}` failed",
+                                branch_name
+                            ));
+                        }
+
+                        let push_status = Command::new("git")
+                            .arg("push")
+                            .arg("origin")
+                            .arg(&branch_name)
+                            .current_dir(workspace)
+                            .status()
+                            .with_context(|| {
+                                format!("Failed to push backport branch {branch_name}")
+                            })?;
+
+                        if !push_status.success() {
+                            return Err(anyhow::anyhow!(
+                                "`git push origin {}` failed",
+                                branch_name
+                            ));
+                        }
+
+                        log::info!(
```

Comment:
We need an extra step here to switch branch back to the original one before we started this whole process so the rest of the post release workflow works.